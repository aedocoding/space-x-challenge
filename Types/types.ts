export interface ILaunchListItem {
  name: string;
  details: string;
  image: string;
}
export type ScreenType = "LaunchList" | "CreateLaunch";
