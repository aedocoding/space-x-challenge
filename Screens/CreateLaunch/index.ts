export{ default as CreateLaunch} from './CreateLaunch'