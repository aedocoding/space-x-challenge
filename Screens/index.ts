export {LaunchDetails} from './LaunchDetails'
export {CreateLaunch} from './CreateLaunch'
export {LaunchList} from './LaunchList'