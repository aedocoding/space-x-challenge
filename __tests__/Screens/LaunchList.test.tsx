import renderer from "react-test-renderer";
import { ScreenType } from "../../Types/types";
import { LaunchList } from "../../Screens";
jest.useFakeTimers();

describe("<LaunchList />", () => {
  const mockNavigatorFunction = (param: ScreenType) => {};
  it("has 1 child", () => {
    const tree = renderer
      .create(<LaunchList navigatorFunction={mockNavigatorFunction} />)
      .toJSON();
    expect(tree.children.length).toBe(1);
  });
  test("renders correctly", () => {
    const tree = renderer
      .create(<LaunchList navigatorFunction={mockNavigatorFunction} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
