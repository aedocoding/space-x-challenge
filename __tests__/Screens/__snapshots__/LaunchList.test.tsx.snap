// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchList /> renders correctly 1`] = `
<Text>
  Loading launches...
</Text>
`;
